### PR TITLE
Merge `--seller-addr` and `--seller-peer-id` into `--seller` parameter

### PR DESCRIPTION
**Pull request by thomaseizinger**
This simplifies the CLI's interface.

It wills also play nicely with https://github.com/comit-network/xmr-btc-swap/pull/593.

### CODE REVIEW

File: swap/src/libp2p_ext.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use libp2p::multiaddr::Protocol;
+use libp2p::{Multiaddr, PeerId};
+
+pub trait MultiAddrExt {
+    fn extract_peer_id(&self) -> Option<PeerId>;
+}
+
+impl MultiAddrExt for Multiaddr {
```
Comment (da-kami) [inline swap/src/libp2p_ext.rs:8]: noice! I will see to reuse this in #593 once merged

File: swap/src/database.rs [hunk 2]
```diff
@@ -110,6 +116,46 @@ impl Database {
         Ok(PeerId::from_str(peer_id.as_str())?)
     }
 
+    pub async fn insert_address(&self, peer_id: PeerId, address: Multiaddr) -> Result<()> {
```
Comment (da-kami) [inline swap/src/database.rs:119]: Did you test this? Could we have a database test that writes and reads addresses?

**Review by da-kami — APPROVED** [review]

**Review by rishflab — APPROVED** [review]

File: swap/src/database.rs (hunk 2 above)
Comment (thomaseizinger) [reply swap/src/database.rs:119]: I tested it by doing a swap and resuming it. Will see to add a test!

Comment (da-kami) [reply swap/src/database.rs:119]: if you can please don't rebase - I based #593 on this

**Pull request merged**